tests/jballerina-unit-test/src/test/resources/test-src/statements/ifelse/type-guard-negative.bal

```diff
@@ -222,3 +222,26 @@ function testMapIntersectionNegative() {
         map<int> x = m;
     }
 }
+
+type CyclicComplexUnion int|CyclicComplexUnion[]|object {};
+type floatUnion float|string;
```

Comment:
Let's add another test where the target type is also a cyclic union. We can do this separately.